balboa-agent/src/main/scala/com/socrata/balboa/agent/BalboaAgent.scala

```diff
@@ -155,4 +97,82 @@ object BalboaAgent extends App with Config with StrictLogging {
     }
   })
 
+
+  // scalastyle:off method.length
+  def amqMetricQueue(): MetricQueue = {
+    logger info "Initializing ActiveMQ connection factory. " +
+      "(This is setting the username, password and destination.)"
+    val amqConnectionFactory = (activemqUser, activemqPassword) match {
+      case (Some(user), Some(password)) => new ActiveMQConnectionFactory(user, password, activemqServer)
+      case _                            => new ActiveMQConnectionFactory(activemqServer)
+    }
+
+    // The ActiveMQ libraries can be obscure when they are misbehaving. The hope
+    // is that by registering this listener, additional information will be
+    // written to the logs for troubleshooting.
+    amqConnectionFactory.setTransportListener(new TransportListener {
```

Comment:
You can shorten this method by moving the TransportListener to it's own concrete class somewhere.

If that isn't enough to get rid of the warning, I see two try blocks down below that can be refactored to methods.